sys/test/test.txt

```diff
@@ -486,6 +486,37 @@ offsetof0 {
 	o7	offsetof[f7, int32]
 }
 
+parent_parent_helper {
+	f1	array[int32, 8]
+	f2	parent_parent_length
+	f3	ptr[in, parent_parent_length]
+}
+
+parent_parent_length {
+	f1	len[parent:parent:f1, int32]
+}
+
+test$length31(a0 ptr[in, parent_parent_helper])
+
+one_parent_length {
+	f1	array[int32, 4]
+	f2	len[parent:f1, int32]
+}
+
+test$length32(a0 ptr[in, one_parent_length])
+
+parent_union_struct {
+	f1	array[int32, 4]
+	f2	ptr[in, parent_union]
+}
+
+parent_union [
+	u1	len[parent:f1, int32]
```

Comment:
The commit description says:
```
Let len[parent:parent:a, T] refer to the structure that encloses the
current one.
```

But this uses just `parent:f1`.

So it's not `parent:parent`, right?
`parent:parent` looks more consistent. Otherwise `parent` in `parent` and in `parent:f` refers to different structs.

Reply:
I've adjusted the code to use `parent:parent:field` for unions as well.